MovieDemo/MovieDemo/Model/Store.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  Store.swift
+//  MovieDemo
+//
+//  Created by Anton Kölbl on 14.01.24.
+//
+
+import Foundation
+
+enum MovieAppError: String, Error {
+  case invalidURL = "URL is invalid"
+  case noData = "No suitable data coming back."
+  case invalidResponse = "Response is invalid"
+}
+
+enum Type: String {
+  case popular = "Popular"
+  case upcoming = "Upcoming"
+}
+
+class Store: ObservableObject {
+  
+  static let apiKey = "5d03645aac4c476d41ab1227e69d1c67"
+  static let baseAddress = "https://api.themoviedb.org/3/"
+  static let popularMoviesAddress = baseAddress + "movie/popular?api_key=\(apiKey)"
+  static let upcomingMoviesAddress = baseAddress + "movie/upcoming?api_key=\(apiKey)"
+  
+  @Published var popularMovies: [Movie] = []
+  @Published var upcomingMovies: [Movie] = []
+  @Published var type = Type.popular
+  
+  var memos: [String: String] = [:]
+  
+  var memoKeys: [String] {
```

Comment:
consider a way to store this (perhaps it is already on your list for the data store requirement). Since this is the only thing you are storing consider writing a function to take the array and convert it to a `Data` type and then use `.write(to:)` so store the data as a file in the documents directory. Then another function using `Data(contentsOf:)` to bring the array back from the documents directory.

Reply:
Look into the environment variable `.scenePhase` to detect when the app moves to the background or foreground (these are the normal places people would read and write this kind of data)